File: programming files/sum_even_fibonacci.cc
#include <iostream>

int main()
{
    // Define first two Fibonacci Numbers
    int a = 0, b = 1;
    int sum_even = 0;
    // Summing even Fibonacci Numbers up to 4000000
    while (b < 4000000) {
        // Check for even number
        if (b % 2 == 0) {
            // Sum if even
            sum_even += b;
        }
        // Update Fibonacci Sequence
        int temp = b;
        b = a + b;
        a = temp;
    }
    // Print the sum
    std::cout << sum_even << std::endl;

    return 0;
}

// answer should be 4613732

File: programming files/sum_even_fibonacci.java
// summing even numbers in the fibonacci sequence
public class sum_even_fibonacci {

    // main function
    public static void main(String[] args) {
        // initializing two integers
        int a = 0, b = 1;
        //sum even initialized to 0
        int sumEven = 0;

        // going through this loop and doing the fibinaci sequence for each one
        while (b < 4000000) {
            if (b % 2 == 0) {
                sumEven += b;
            }
            int temp = b;
            b = a + b;
            a = temp;
        }
        // printing the number
        System.out.println(sumEven);
    }
}

// answer should be 4613732
